java-checks/src/test/files/checks/RedundantThrowsDeclaration.java

```diff
@@ -58,3 +58,105 @@ public String plop() {
     }
   }
 }
+
+abstract class MySuperClass {
+  abstract foo() throws MyException;
+}
+
+abstract class ThrownCheckedExceptions extends MySuperClass {
+  public ThrownCheckedExceptions(String s) throws MyException { // Noncompliant {{Remove the declaration of thrown exception 'MyException', as it cannot be thrown from constructor's body.}}
+    bar();
+  }
+
+  @Override
+  void foo() throws MyException { //Compliant - Override
+    bar();
+  }
+
+  void foo0() throws MyException { // Noncompliant {{Remove the declaration of thrown exception 'MyException', as it cannot be thrown from method's body.}}
+    bar();
+  }
+
+  void foo1() throws MyException { // Compliant - unknown method is called
+    unknownMethod();
+    bar();
+  }
+
+  void foo2() throws MyException { // Compliant - unknown exception is thrown by qix
+    qix();
+  }
+
+  void foo3() throws MyException { // Noncompliant {{Remove the declaration of thrown exception 'MyException', as it cannot be thrown from method's body.}}
+    new MyClass() {
+      void foo() {
+        throw new MyException();
+      }
+    };
+  }
+
+  void foo4() throws MyException { // Noncompliant
+    gul(o -> { throw new RuntimeException(); });
+  }
+
+  void foo5() throws MyException { // Compliant
+    bar();
+    mok();
+  }
+
+  void foo6() { // Compliant
+    puf();
+    kal();
+  }
+
+  void foo7() throws MyException { // Compliant - False-Negative - studying control flow is not done
+    try {
+      mok();
+    } catch(MyException e) {
+      // do nothing
+    }
+  }
+
+  void foo8(Throwable t) throws java.io.IOException { // Compliant - kill the noise due to SONARJAVA-1778
```

Comment:
Again : I tend to think that : false negative is better suited than kill the noise for comments in code.